fluentd/vendored_gem_src/syslog_protocol/lib/syslog_protocol/packet.rb

```diff
@@ -23,6 +40,15 @@ def assemble(max_size = 1024)
       data
     end
 
+    def assemble_rfc5424()
+      unless @hostname and @facility and @severity and @appname
+        raise "Could not assemble packet without hostname, appname, facility, and severity"
+      end
+      data = "<#{pri}>1 #{generate_timestamp_rfc5424} #{@hostname} #{@appname} #{@procid} #{@msgid} #{structured_data} #{@content}"
```

Comment:
ruby always returns the last line so you can return the formatted string

Reply:
updated

Reply:
not expecting an actual change here but you don't even need to do variable assignment here.  ruby will return the result of the last line